javascript/javascript1/week2/homework-week2.js

```diff
@@ -0,0 +1,126 @@
+//Flight booking fullname function
+
+function getFullname (firstname, surname){
+    return firstname + " " + surname;
+}
+var fullname1 = getFullname("Benjamin", "Hughes");
+var fullname2 = getFullname("Atefeh", "Behroozi");
+
+console.log(fullname1);
+console.log(fullname2);
+
+
+
+
+
+
+
+
+//Formal fullname
+function getFullname(firstname, surname, useFormalName = false) {
+    if (firstname === "" || surname === "") {
+      return "Invalid input, both firstname and surname are required.";
+    }
+    if (useFormalName) {
+      return "Lord " + firstname + " " + surname;
+    } else {
+      return firstname + " " + surname;
+    }
+  }
+
+var fullname1 = getFullname("Benjamin", "Hughes", true);
+var fullname2 = getFullname("Atefeh", "Behroozi");
+
+console.log(fullname1); 
+console.log(fullname2); 
+
+
+
+
+
+
+
+
+//Event application
+
+function getEventWeekday(days) {
+    const weekdays = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+    const today = new Date().getDay();
+    const eventDay = (today + days) % 7;
+    return weekdays[eventDay];
+   
+  }
+  console.log(getEventWeekday(5));
+  console.log(getEventWeekday(9)); 
+  console.log(getEventWeekday(2)); 
+
+
+
+
+
+
+   
+
+//Weather wear
+
+function recommendClothing(temperature) {
+    if (temperature >= 30) {
```

Comment:
check if the switch condition can make this more optimize.

Reply:
I write these switch codes:
function recommendClothing(temperature) {
  switch (true) {
    case temperature >= 30:
      return "shorts and a tank top";
    case temperature >= 20:
      return "shorts and a t-shirt";
    case temperature >= 10:
      return "jeans and a sweater";
    default:
      return "a coat and warm pants";
  }
}

const clothesToWear = recommendClothing(18);
console.log(clothesToWear);